tests/test_unit/test_image_mosaic.py

```diff
@@ -127,3 +148,145 @@ def test_data_for_napari(image_mosaic, test_constants):
     ):
         assert tile_data[0].shape == test_constants["TILE_SIZE"]
         assert (tile_data[1] == expected_pos).all()
+
+
+@pytest.mark.parametrize(
+    "output_file_name, fuse_function_name",
+    [
+        ("fused.zarr", "_fuse_to_zarr"),
+        ("fused.zarr", "_fuse_to_zarr"),
+        ("fused.zarr", "_fuse_to_zarr"),
+        ("fused.zarr", "_fuse_to_zarr"),
+        ("fused.h5", "_fuse_to_bdv_h5"),
+        ("fused.h5", "_fuse_to_bdv_h5"),
+        ("fused.h5", "_fuse_to_bdv_h5"),
+        ("fused.h5", "_fuse_to_bdv_h5"),
+    ],
+)
+def test_fuse(
+    image_mosaic, mocker, output_file_name, fuse_function_name, test_constants
+):
+    mock_fuse_function = mocker.patch(
+        f"brainglobe_stitch.image_mosaic.ImageMosaic.{fuse_function_name}",
+    )
+
+    image_mosaic.fuse(output_file_name)
+
+    default_downscale_factors = (1, 2, 2)
+    mock_fuse_function.assert_called_once_with(
+        image_mosaic.xml_path.parent / output_file_name,
+        test_constants["EXPECTED_FUSED_SHAPE"],
+        downscale_factors=default_downscale_factors,
+    )
+
+
+@pytest.mark.parametrize(
+    "pyramid_depth, num_channels",
+    [(1, 1), (1, 2), (1, 5), (2, 1), (2, 2), (2, 5), (3, 1), (3, 2), (3, 5)],
+)
+def test_get_metadata_for_zarr(image_mosaic, pyramid_depth, num_channels):
+    backup_num_channels = image_mosaic.num_channels
+    image_mosaic.num_channels = num_channels
+
+    metadata, axes = image_mosaic.get_metadata_for_zarr(pyramid_depth)
+
+    if num_channels > 1:
+        assert len(axes) == 4
+        assert axes[0]["name"] == "c"
+        assert axes[0]["type"] == "channel"
+        axes.pop(0)
+    else:
+        assert len(axes) == 3
+
+    expected_axes_names = ["z", "y", "x"]
+
+    for idx, axes in enumerate(axes):
+        assert axes["name"] == expected_axes_names[idx]
+        assert axes["type"] == "space"
+        assert axes["unit"] == "micrometer"
+
+    for idx, transformation in enumerate(metadata):
+        assert transformation[0]["type"] == "scale"
+        expected_scale = [
+            image_mosaic.z_resolution,
+            image_mosaic.x_y_resolution * 2**idx,
+            image_mosaic.x_y_resolution * 2**idx,
+        ]
+        if num_channels > 1:
+            expected_scale.insert(0, 1)
+
+        assert transformation[0]["scale"] == expected_scale
+
+    image_mosaic.num_channels = backup_num_channels
+
+
+def test_fuse_to_zarr(image_mosaic, test_constants):
+    pyramid_depth = 3
```

Comment:
I wanted to, however since the test image is substantially smaller than real data, you quickly end up in a situation where one of the axes has a size of 1 leading to undefined behaviour. 3 was the most I could!